src/sparse/KokkosSparse_sptrsv_supernode.hpp

```diff
@@ -1427,17 +1443,26 @@ invert_supernodal_columns(KernelHandle *kernelHandle, bool unit_diag, int nsuper
         timer.reset ();
         #endif
         if(run_trmm_on_device) {
+          Kokkos::View<scalar_t**, Kokkos::LayoutLeft, trmm_memory_space, Kokkos::MemoryUnmanaged>
+            dViewL (trmm_dwork.data(), nsrow, nscol);
+
+          #if !defined(KOKKOSKERNELS_ENABLE_TPL_MAGMA)
```

Comment:
should this be removed? perhaps add a runtime error check for:
```
#if !defined(KOKKOSKERNELS_ENABLE_TPL_CUBLAS)
if(run_trmm_on_device)
  // throw error
#endif
```

Reply:
Thank you for the review, @e10harvey!!

I was not clear about this comment. Right now, if `!defined(KOKKOSKERNELS_ENABLE_TPL_CUBLAS)`, then run_trmm_on_device is always false. So, if the user asked to run big TRMMs on device, then we just ignore it (or consider all to be too small for `KokkosBlas::trmm`). You are suggesting to throw an error instead?

Reply:
I see. Thanks for clarifying. It looks like the trmm & trtri fallback implementations for device views are causing maintenance issues here in sptrsv; would it help to update the implementations such that the checks for `KOKKOS_ENABLE_TPL_CUBLAS` and `KOKKOS_ENABLE_TPL_MAGMA` are not needed here? If so, this can be addressed via #1046 and #1047.

Reply:
Thank you, @e10harvey, and you are right. If Kokkos-Kernels TRMM & TRTRI perform well and can avoid the deep-copy to the host, then I think we won't need the checks.